### PR TITLE
gppkg: Fix gppkg --remove name matching regex

### PR DESCRIPTION
**Pull request by nadeemg**
When a user has multiple gppkgs installed, and they try to run gppkg
--remove, we were actually giving back all of the packages instead of
the specific one that the user wants to remove.

Tighten up our regex to match only the name that the user specified in
the argument.

Make it so that gppkg --remove will now do an equality check <package
name>.gppkg before trying out the previous behavior. Message the user if
there is more than one match.

Signed-off-by: Marbin Tan <mtan@pivotal.io>

### CODE REVIEW

File: gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py [hunk 1]
```diff
@@ -15,11 +14,13 @@ def setUp(self):
...
-            patch('gppylib.programs.gppkg.ListFilesByPattern', return_value=self.mock_list_files_by_pattern),
             patch('gppylib.programs.gppkg.Gppkg', return_value=self.mock_gppkg),
             patch('gppylib.programs.gppkg.UninstallPackage', return_value=self.mock_uninstall_package),
+            patch('os.listdir')
         ])
         self.mock_logger = self.get_mock_from_apply_patch('logger')
+        self.mock_listdir = self.get_mock_from_apply_patch('listdir')
+
 
     def tearDown(self):
```
Comment (nadeemg) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py]: why is this teardown listed explicitly here?  isn't it the default?

File: gpMgmt/bin/gppylib/programs/gppkg.py [hunk 2]
```diff
@@ -212,11 +212,19 @@ def run(self):
             query_type, package_path = self.query
             QueryPackage(query_type, package_path).run()
         elif self.remove:
-            pkg_file_list = ListFilesByPattern(GPPKG_ARCHIVE_PATH, '*' + GPPKG_EXTENSION).run()
+            pkg_file_list = ListFilesByPattern(GPPKG_ARCHIVE_PATH, self.remove + GPPKG_EXTENSION).run()
 
+            # Check for equality first, then let's be more 'vague' with what we can remove.
```
Comment (larham) [inline gpMgmt/bin/gppylib/programs/gppkg.py]: this comment seems to be about the pattern chosen on line 215, so consider moving it up above that.

File: gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py [hunk 3]
```diff
@@ -55,3 +57,24 @@ def test__remove_succeeds_when_gppkg_had_been_installed(self):
 
         self.mock_list_files_by_pattern.run.assert_called_once()
         self.mock_uninstall_package.run.assert_called_once()
+
```
Comment (larham) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py]: how about a test for an exact match, when there are 2 that would match a "*" test?  e.g., the word "sample" when there are packages "sample" and "sample2"?

File: gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py (hunk 1 above)
Comment (larham) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py]: done

File: gpMgmt/bin/gppylib/programs/gppkg.py (hunk 2 above)
Comment (nadeemg) [inline gpMgmt/bin/gppylib/programs/gppkg.py]: done

File: gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py (hunk 3 above)
Comment (nadeemg) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gppkg.py]: added test

**Comment by litzell** [comment]
Added a release note in the Fixes section:
http://docs-pivotal-gpdb-staging.cfapps.io/500Beta/relnotes/GPDB_500_README.html#topic_ukk_rxm_b1b

**Pull request merged**